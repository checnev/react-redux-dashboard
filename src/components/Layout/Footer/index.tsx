import {
  Container,
  Typography,
  Divider,
} from '@material-ui/core';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {
    marginTop: 'auto',
  },
  container: {
    paddingTop: 20,
    paddingBottom: 20,
  },
  policy: {
    fontSize: '.9rem',
  },
}));

const Footer = () => {
  const classes = useStyles();
  const year = new Date().getFullYear();

  return (
    <footer className={classes.root}>
      <Divider />
      <Container maxWidth="lg" className={classes.container}>
        <Typography align="right">CarShare Â© {year}</Typography>
      </Container>
    </footer>
  );
};

export default Footer;
