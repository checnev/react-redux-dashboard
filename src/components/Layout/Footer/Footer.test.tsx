import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Footer from 'components/Layout/Footer';

let component: any = null;
beforeEach(() => {
  component = document.createElement('div');
  document.body.append(component);
});

afterEach(() => {
  unmountComponentAtNode(component);
  component.remove();
  component = null;
});

describe('Footer', () => {
  it('should render', () => {
    act(() => {
      render(<Footer/>, component);
    });

    expect(
      pretty(component.innerHTML)
    ).toMatchSnapshot();
  });
});
