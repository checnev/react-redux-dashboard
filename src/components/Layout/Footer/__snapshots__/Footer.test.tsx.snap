// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
"<footer class=\\"makeStyles-root-1\\">
  <hr class=\\"MuiDivider-root\\">
  <div class=\\"MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthLg\\">
    <p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-alignRight\\">CarShare Â© 2021</p>
  </div>
</footer>"
`;
